### ISSUE TITLE
Intel 19.1 and 19.0 constexpr error

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#12507]
9.3 and master fails with
```
 /home1/08067/msmaier/workspace/dealii/include/deal.II/base/geometry_info.h(2613): error: expression must have a constant value
         internal::GeometryInfoHelper::Initializers<dim>::unit_normal_vector();
         ^
 Note: too few arguments in function call
           detected during:
             instantiation of class "dealii::GeometryInfo<dim> [with dim=3]" at line 330 of "/home1/08067/msmaier/workspace/dealii/source/fe/mapping_cartesian.cc"
             instantiation of "void dealii::MappingCartesian<dim, spacedim>::maybe_update_normal_vectors(unsigned int, const dealii::MappingCartesian<dim, spacedim>::InternalData &, std::vector<dealii::Tensor<1, dim, double>, std::allocator<dealii::Tensor<1, dim, double>>> &) const [with dim=3, spacedim=3]" at line 480 of "/home1/08067/msmaier/workspace/dealii/source/fe/mapping_cartesian.cc"
```
As reported by @tamiko and @zjiaqi2018 

@tamiko has narrowed it down to a single patch. Can you share?

**Comment by tjhei** [comment]
The problem is 6353b518b2b0918940c8776037f2c0610466d66
?

**Comment by tamiko** [comment]
@tjhei Yes, the issue is precisely this commit. I can confirm that reverting (the entire changeset in the pull request in question) fixes the compilation issue with intel 19.1.

**Comment by tamiko** [comment]
@tjhei I will test tomorrow whether gcc-11 still needs that patchset, or wether gcc's behavior has been changed/fixed.
